fhir-server/src/main/java/com/ibm/fhir/server/resources/Capabilities.java

```diff
@@ -71,54 +74,83 @@
+    private static final String ERROR_CONSTRUCTING = "An error occurred while constructing the Conformance statement.";
+
+    // Capability Statement Cache per Tenant
+    private static ConcurrentHashMap<String, CapabilityStatement> CAPABILITY_STATEMENT_CACHE_PER_TENANT =
+            new ConcurrentHashMap<>();
+
+    // Constructor
+    public Capabilities() throws Exception {
+        super();
+    }
+
     @GET
     @Path("metadata")
-    public Response capabilities() throws ClassNotFoundException {
-        log.entering(this.getClass().getName(), "metadata()");
-        Date startTime = new Date();
-        String errMsg = "Caught exception while processing 'metadata' request.";
-
+    public Response capabilities() {
+        log.entering(this.getClass().getName(), "capabilities()");
         try {
+            Date startTime = new Date();
             checkInitComplete();
-
-            CapabilityStatement capabilityStatement = getCapabilityStatement();
+            CapabilityStatement capabilityStatement = getOrCreateCapabilityStatement();
             RestAuditLogger.logMetadata(httpServletRequest, startTime, new Date(), Response.Status.OK);
 
             return Response.ok().entity(capabilityStatement).build();
         } catch (FHIROperationException e) {
-            log.log(Level.SEVERE, errMsg, e);
+            log.log(Level.SEVERE, ERROR_MSG, e);
             return exceptionResponse(e, issueListToStatus(e.getIssues()));
         } catch (Exception e) {
-            log.log(Level.SEVERE, errMsg, e);
+            log.log(Level.SEVERE, ERROR_MSG, e);
             return exceptionResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
         } finally {
-            log.exiting(this.getClass().getName(), "metadata()");
+            log.exiting(this.getClass().getName(), "capabilities()");
+        }
+    }
+
+    /*
+     * get or create capability statement
+     */
+    private CapabilityStatement getOrCreateCapabilityStatement() throws FHIROperationException {
+        // Get TenantId
+        FHIRRequestContext ctx = FHIRRequestContext.get();
+        String tenantId = ctx.getTenantId();
+        CapabilityStatement statement = CAPABILITY_STATEMENT_CACHE_PER_TENANT.get(tenantId);
+        if (statement == null) {
+            statement = getCapabilityStatement();
+            CAPABILITY_STATEMENT_CACHE_PER_TENANT.put(tenantId, statement);
+        } else {
+            // Previously the Conformance Statement was built
+            // using ZonedDateTime.now(ZoneOffset.UTC)
+            TemporalAccessor acc = statement.getDate().getValue();
+            ZonedDateTime cachedTime = ZonedDateTime.from(acc);
+
+            if (ZonedDateTime.now().isBefore(cachedTime.plusHours(1))) {
+                statement = getCapabilityStatement();
+            }
         }
+        return statement;
     }
 
     private synchronized CapabilityStatement getCapabilityStatement() throws FHIROperationException {
```

Comment:
I've always wondered why this method is synchronized.  Definitely orthogonal to your PR, but let me know if you agree it shouldn't be needed...

Reply:
I think you are going to want to use the compute method on ConcurrentHashMap

Reply:
and computeIfAbsent for the first one

Reply:
Let me remove. If we find we need to add it back we can.

Reply:
what about something like:
```
CAPABILITY_STATEMENT_CACHE.compute(tenantId, k -> getOrCreateCapabilityStatement());
```

Reply:
compute should be atomic (for ConcurrentHashMap)

Reply:
and compute returns the new value:
```
CapabilityStatement capabilityStatement = CAPABILITY_STATEMENT_CACHE.compute(tenantId, k -> getOrCreateCapabilityStatement(tenantId));
```

Reply:
Flipped around per conversation